src/db-adapters/Mongoose/MongooseAdapter.js

```diff
@@ -603,30 +604,37 @@ export default class MongooseAdapter {
     return words.join(" ");
   }
 
-  static getIdQueryType(idOrIds) {
-    const mode = typeof idOrIds === "string" ? "findOne" : "find";
-    let idQuery;
+  static getIdQueryType(idOrIds, Model) {
+    if (idOrIds == null) {
+      return Q([ "find", undefined ]);
+    }
+
+    const [ mode, ids ] = Array.isArray(idOrIds) ? [ "find", idOrIds ] : [ "findOne", [ idOrIds ] ];
+    const idValidationPromises = ids.map(id => this.validateId(id, Model));
 
-    if (typeof idOrIds === "string") {
-      if (!this.idIsValid(idOrIds)) {
+    return Q.all(idValidationPromises).then(() => {
+      const idQuery = { _id: mode === "find" ? { $in: ids } : ids[0] };
+      return [ mode, ids.length ? idQuery : undefined ];
```

Comment:
Are you sure this `ids.length ? idQuery : undefined` ternary is needed? 

If I read correctly, it means that calling `getIdQueryOrType` with `[]` will resolve to `["find", undefined]`, which I don't think is what we want, since I'd interpret an array of ids—even if it's empty—to be about returning docs whose ids are from that array, rather than interpreting it as a findAll query. (This seems pedantic, but I can imagine cases where it might come up.)